db/snapshot/backup_task.cc

```diff
@@ -65,18 +64,29 @@ tasks::is_abortable backup_task_impl::is_abortable() const noexcept {
 }
 
 future<tasks::task_manager::task::progress> backup_task_impl::get_progress() const {
-    auto ret = _total_progress;
-    ret.completed = _progress.uploaded;
-    co_return ret;
+    co_await update_total_progress();
```

Comment:
This is called via API, and `co_await _sharded_worker.stop();` is called by task itself. If it happens in parallel, won't `update_total_progress()`'s call of map_reduce0 step on stopping `_sharded_worker` thing?

Reply:
There's another approach. The sstable_loader code also has sharded progress, and it solves this by carefully maintaining state of it, keeping the final map-reduced value when its stopped

Reply:
I just checked and yes, there is the same bug there:

https://github.com/scylladb/scylladb/blob/84d4af1f0ef67bdbdcc49c4f5d4f53b7ef14ea81/sstables_loader.cc#L638-L668

if API invoked get_progress preempts in line 656 and release_resources() is started

Reply:
@bhalevy , std::vector<progress> as in the recent version should work as well (not very cache friendly, but it's not updated that often)

Reply:
https://github.com/scylladb/scylladb/issues/23801 not to forget